non-inlined function doesn't fail go:inline directive

Repro steps:

Check out github.com/cockroachdb/pebble@78d53457

Apply this diff:
```
diff --git a/internal/invariants/invariants.go b/internal/invariants/invariants.go
index 2a0cee5e..f2d1d733 100644
--- a/internal/invariants/invariants.go
+++ b/internal/invariants/invariants.go
@@ -25,6 +25,8 @@ const RaceEnabled = buildtags.Race
 // Sometimes returns true percent% of the time if invariants are Enabled (i.e.
 // we were built with the "invariants" or "race" build tags). Otherwise, always
 // returns false.
+//
+//gcassert:inline
 func Sometimes(percent int) bool {
        return Enabled && rand.Uint32N(100) < uint32(percent)
 }
```

Run GCAssert:
```
$ go test ./internal/lint/ -v -run  /GCAssert
=== RUN   TestLint
=== RUN   TestLint/TestGCAssert
=== PAUSE TestLint/TestGCAssert
=== CONT  TestLint/TestGCAssert
    lint_test.go:135: See /var/folders/x0/5yrxw_4s271g705vjscld23c0000gq/T/gcassert-3766173111.log for full output.
--- PASS: TestLint (0.53s)
    --- PASS: TestLint/TestGCAssert (0.79s)
PASS
```

Full output starts with:
```
/Users/radu/go/go1.22.7/bin/go build -gcflags=-m=2 -d=ssa/check_bce/debug=1 ././cockroachkvs ././internal/invariants ././internal/lint ././sstable/colblk
# github.com/cockroachdb/pebble/internal/invariants
internal/invariants/invariants.go:30:6: cannot inline Sometimes: function too complex: cost 83 exceeds budget 80
internal/invariants/invariants.go:31:32: inlining call to rand.Uint32N
internal/invariants/invariants.go:31:32: inlining call to rand.(*Rand).Uint32N
```

--- a/gcassert.go
+++ b/gcassert.go
@@ -390,10 +390,11 @@ func (v *inlinedDeclVisitor) Visit(node ast.Node) (w ast.Visitor) {
 			obj = v.p.TypesInfo.Uses[n]
 		case *ast.SelectorExpr:
 			sel := v.p.TypesInfo.Selections[n]
-			if sel == nil {
-				break
+			if sel != nil {
+				obj = sel.Obj()
+			} else {
+				obj = v.p.TypesInfo.Uses[n.Sel]
 			}
-			obj = sel.Obj()
 		}
 		if _, ok := v.mustInlineFuncs[obj]; ok {
 			lineInfo := v.directiveMap[lineNumber]

--- a/gcassert_test.go
+++ b/gcassert_test.go
@@ -99,13 +99,14 @@ testdata/inline.go:51:	sum += notInlinable(i): call was not inlined
 testdata/inline.go:55:	sum += 1: call was not inlined
 testdata/inline.go:58:	test(0).neverInlinedMethod(10): call was not inlined
 testdata/inline.go:60:	otherpkg.A{}.NeverInlined(sum): call was not inlined
+testdata/inline.go:62:	otherpkg.NeverInlinedFunc(sum): call was not inlined
 testdata/issue5.go:4:	Gen().Layout(): call was not inlined
 `
 
 	testCases := []struct{
-		name string
-		pkgs []string
-		cwd string
+		name     string
+		pkgs     []string
+		cwd      string
 		expected string
 	}{
 		{
@@ -130,7 +131,7 @@ testdata/issue5.go:4:	Gen().Layout(): call was not inlined
 				"./testdata",
 				"./testdata/otherpkg",
 			},
-			cwd: cwd,
+			cwd:      cwd,
 			expected: expectedOutput,
 		},
 	}

--- a/testdata/inline.go
+++ b/testdata/inline.go
@@ -58,4 +58,6 @@ func caller() {
 	sum += test(0).neverInlinedMethod(10)
 
 	otherpkg.A{}.NeverInlined(sum)
+
+	otherpkg.NeverInlinedFunc(sum)
 }

--- a/testdata/otherpkg/foo.go
+++ b/testdata/otherpkg/foo.go
@@ -10,3 +10,10 @@ func (a A) NeverInlined(n int) {
 		fmt.Println(i)
 	}
 }
+
+//gcassert:inline
+func NeverInlinedFunc(n int) {
+	for i := 0; i < n; i++ {
+		fmt.Println(i)
+	}
+}

